### DISCUSSION TITLE
DPM8624 from Joy-IT connected to iobroker

### DISCUSSION THREAD

**Discussion by io-taster** [arendst/Tasmota#17960] [Q&A]
I try to connect the DPM8624 to iobroker.
The bridge is an ESP32, flashed with tasmoto scripting.
The script comes from github

radiergimmy
Nov 29, 2022

I'm using this for my DPM8624

>D
Vmax=60 ;max value from specifications https://joy-it.net/en/products/JT-DPM8624
Cmax=24 ;max value from specifications https://joy-it.net/en/products/JT-DPM8624
v=0 ;store device voltage state (targeted)
V=0 ;store WebUI voltage state (targeted)
c=0 ;store device current state (targeted)
C=0 ;store WebUI current state (targeted)
sw=0 ;store device switch state
SW=0 ;store WebUI switch state
rV="01060000" ;ModBus register (voltage)
vV="" ;store hex value of V to append to rV
rC="01060001" ;ModBus register (current)
vC="" ;store hex value of C to append to rC
rSW="01060002000" ;ModBus register (switch)
vSW="" ;store hex value of SW to append to rSW

>B
smlj=0 ;disable publishing at MQTT teleperiod, on boot
->sensor53 r

>R
smlj=0 ;disable publishing at MQTT teleperiod, on reboot

>S
if upsecs>30
then
smlj=1 ;re-enable publishing at MQTT teleperiod, after boot
endif

>W
bu(SW "DPM ON" "DPM OFF") ;toggle switch via WebUI
nm(0.0 60.0 0.01 V "DPM out (V)" 200 2) ;set voltage output via WebUI
nm(0.0 24.0 0.01 C "DPM out (A)" 200 2) ;set current output via WebUI

>T
sw=DC#sSwitch ;load switch state from ModBus variable >M
v=DC#sVolt ;load voltage state from ModBus variable >M
c=DC#sCur ;load voltage state from ModBus variable >M

if ((chg[sw]>0) and (SW!=sw)) ;update WebUI based on physical input (switch)
then
  SW=sw
endif

if ((chg[v]>0) and (V!=v)) ;update WebUI based on physical input (voltage)
then
  V=v
endif

if ((chg[c]>0) and (C!=c)) ;update WebUI based on physical input (current)
then
  C=c
endif

if chg[V]>0 ;update ModBus register based on WebUI input (voltage)
then
  if V>Vmax ;prevent out of bound value (voltage max)
  then
    V=Vmax
  endif
  if V<0 ;prevent out of bound value (voltage min)
  then
    V=0
  endif
  vV=rV+sb(hx((V*100)) 4 4)
  sml(1 3 vV)
endif

if chg[C]>0 ;update ModBus register based on WebUI input (current)
then
  if C>Cmax ;prevent out of bound value (current max)
  then
    C=Cmax ;prevent out of bound value (current min)
  endif
  if C<0
  then
    C=0
  endif
  vC=rC+sb(hx((C*1000)) 4 4)
  sml(1 3 vC)
endif

if ((chg[SW]>0) and (sw!=SW) ;update ModBus register based on WebUI input (switch)
then
  vSW=rSW+s(SW)
  sml(1 3 vSW)
endif

>M 1
+1,4,m,16,9600,DC,5,2,010300000001,010300010001,010300020001,010310010001,010310020001 ;ModBus register taken from JT-DPM86XX_Communication_protocol_2022-07-06.pdf provided by Joy-IT
1,010302SSssxxxx@i1:100,Voltage (set),V,sVolt,2
1,010302SSssxxxx@i2:1000,Current (set),A,sCur,2
1,010302SSssxxxx@i3:1,Switch (set),,sSwitch,0
1,010302SSssxxxx@i4:100,Voltage (disp),V,dVolt,2
1,010302SSssxxxx@i5:1000,Current (disp),A,vCur,2
#

GPIO 1 is SerBr TX
GPIO 3 is SerBr RX
There i can set the Volt and Current

The ESP32 is connected by MQTT to iobroker and works.

The problem is: The tasmata console and iobroker show only value from zero.

![1](https://user-images.githubusercontent.com/124024833/219488741-0ad03994-1ccb-4678-b03e-c5ca5f218cc7.jpg)
![2](https://user-images.githubusercontent.com/124024833/219488766-249f16e4-957d-4bf5-a055-831a4cd852ce.jpg)

how do I get the set values displayed?

**Comment by MatheoM85** [comment]
Hi,
I have the same problem. And I find no answers for that. Do you fix the problem?

**Comment by Gamalinger** [comment]
Hello all,
could somebody help me please?
I get no information from DPM8624 there are only 0 and it is not possible to send some information.
The correct firmware should be installed and the code is inside my Node MCU ESP8266. Everything looks fine but it diplay only 0.
I have other devices connected to NodeRed and the most will be shown by Grafana. Inside NodeRed will be shown only 0 as well.
